dvc/remote/gdrive.py

```diff
@@ -195,10 +212,40 @@ def drive(self):
 
             # Pass non existent settings path to force DEFAULT_SETTINGS loading
             gauth = GoogleAuth(settings_file="")
-            gauth.CommandLineAuth()
 
-            if os.getenv(RemoteGDrive.GDRIVE_USER_CREDENTIALS_DATA):
-                os.remove(self.gdrive_user_credentials_path)
+            try:
+                gauth.CommandLineAuth()
+            except RefreshError as e:
+                self.wipe_out_user_credentials()
+                raise GDriveAccessTokenRefreshError(
+                    msg="Google Drive's access token "
+                    "refreshment is failed: {}".format(str(e))
+                )
+            except KeyError as e:
+                self.wipe_out_user_credentials()
+                raise GDriveMissedCredentialKeyError(
+                    msg='Google Drive\'s user credentials file "{}" '
+                    "misses value for key {}".format(
+                        self.gdrive_user_credentials_path, str(e)
+                    )
+                )
+            except AuthenticationError as e:
+                self.wipe_out_user_credentials()
+                raise DvcException(
+                    "Google Drive authentication process is failed: {}".format(
+                        str(e)
```

Comment:
Same here, you could simply
```suggestion
                        cause=e,
```

Reply:
thanks!
I've left as is only for `KeyError` exceptions, since original message doesn't contain enough details to understand the cause.